source/parser/declaration.civet

```diff
@@ -264,10 +266,13 @@ function processDeclarationConditionStatement(s: IfStatement | IterationStatemen
   { condition } := s
   return unless condition?.expression
   { expression } .= condition
+
   // Support for negated conditions built by unless/until
-  switch expression
-    {type: 'UnaryExpression', children: ['!', {type: 'ParenthesizedExpression', expression: expression2}]}
-      expression = expression2
+  negated .= false
+  if {type: 'UnaryExpression', children: ['!', {type: 'ParenthesizedExpression', expression: expression2}]} := expression
```

Comment:
We may want to add the negated flag as a property on the `IfStatement` node to distinguish between

`if !(x)` and `unless x`